Update scrapy & dependencies

teh opened the pull request:
###### Motivation for this change


###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

teh committed:
parsel: 1.0.3 -> 1.1.0

teh committed:
w3lib: 1.14.2 -> 1.17.0

teh committed:
scrapy: 1.1.2 -> 1.3.1

domenkozar commented:
cc @jskrablin

FRidh reviewed (approved):
haven't tested it but it looks good

teh commented:
@drewkett you ok with these changes?

drewkett commented:
Looks fine to me, though I'd probably add a comment to the nix file explaining why the permissions patch is needed rather than having that information only be in a commit message.

I'm not using this package at the moment, so feel free to take over as maintainer if you'd like.

teh committed:
Move scrapy to its own module and add patch to fix broken permission code.

Scrapy is usually installed via pip where copying all permissions
makes sense. In Nix the files copied are owned by root and
readonly. As a consequence scrapy can't edit the project templates so

  scrapy startproject

fails.

teh commented:
I left you in as the maintainer for now - I'm not directly using them either, had some unrelated code that needed a newer scrapy. 

Updated according to your comment.

teh commented:
@FRidh - I think this is ready to merge.

FRidh commented on pkgs/development/python-modules/scrapy/default.nix:
```diff
@@ -0,0 +1,36 @@
+{ self, buildPythonPackage, pkgs, lib }:
```
- never pass in `pkgs`
- python libraries are passed the individual python packages, not pythonPackages or self

teh replied:
done, PTAL

FRidh commented:
thanks
